### PR TITLE
BN#15 feat(view) view specific item

### PR DESCRIPTION
**Pull request by nellykawera**
included productController
included productRoutes
included swagger documentation
[finish #15]

# PR Description
A buyer should be able to see any item from the available items.
# Description of tasks that were expected to be completed
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [v] Create Given a get request to retrieve the details of a specific item in a sellers collection
# Description of tasks that were expected to be completed and that are not yet completed
- [v] None
# How this may be tested?
Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
1. run `npm i` to install all required dependencies
2. run `npm run dev`
3. Swagger/Get
# Screenshots (If appropriate/applicable)
Provide screenshots related to your PR in case it's appropriate and applicable
# Please check this Checklist before you submit your PR:
- [v] My code follows the style guidelines of this project
- [v] I have performed a self-review of my code
- [v] I have commented on my code, particularly in hard-to-understand areas
- [v] My code generates no warnings
- [v] There are no vulnerabilities

### CODE REVIEW

File: src/index.js
Comment (dushimeemma) [inline src/index.js:1]: You didn't rebase @nellykawera

File: src/tests/sum.test.js
Comment (dushimeemma) [inline src/tests/sum.test.js:1]: What is the use of this file with this task?

File: src/controllers/productController.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  try {
+    const { id } = req.params;
+    const specificItem = await product.findOne({
+      where: { id },
+    });
+
+    if (!id) {
+      return res.status(404).json({
+        status: 'fail',
+        message: 'Missing id parameter in request.',
```
Comment (EvelyneUmubyeyi) [inline src/controllers/productController.js]: Could you use translation on all the messages returned in the API response.

File: src/controllers/productController.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      });
+    }
+
+    if (!specificItem) {
+      return res.status(404).json({
+        status: 'fail',
```
Comment (EvelyneUmubyeyi) [inline src/controllers/productController.js]: You can change status: 'fail' to status: 404

File: src/controllers/productController.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        message: 'You are not allowed to perform this operation',
+      });
+    }
+
+    res.status(200).json({
+      status: 'success',
+      item: specificItem,
+    });
+  } catch (err) {
+    res.status(400).json({
```
Comment (EvelyneUmubyeyi) [inline src/controllers/productController.js]: I think status code should be 500 here.

File: src/controllers/productController.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      status: 'fail',
+      message: err.message,
+    });
+  }
+};
+
```
Comment (EvelyneUmubyeyi) [inline src/controllers/productController.js:50]: I think another endpoint specifically for when a seller wants to retrieve a specific item from their own collection would be needed.

**Review by EvelyneUmubyeyi — CHANGES_REQUESTED** [review]

File: src/controllers/productController.js (hunk 1 above)
Comment (Charlesntwari) [reply src/controllers/productController.js]: good work @nellykawera  you putted your effort in this project  but it's also a good practice to not push console's, so i would suggest you to remove them

File: src/tests/sum.test.js
Comment (nellykawera) [reply src/tests/sum.test.js:1]: this file is not mine

File: src/database/seeders/20230326221727-demo-cogito.js [hunk 5]
```diff
@@ -98,7 +98,15 @@ module.exports = {
...
         {
+<<<<<<< HEAD
           user_id: 3,
+=======
+<<<<<<< HEAD
+          user_id: 2,
+=======
+          user_id: 3,
+>>>>>>> 6da50f4 (* feat(profile) create profile feature (#37))
+>>>>>>> e257e9b (* feat(profile) create profile feature (#37))
```
Comment (Niyobuzima) [inline src/database/seeders/20230326221727-demo-cogito.js]: delete this line

File: src/database/seeders/20230326221727-demo-cogito.js [hunk 6]
```diff
@@ -98,7 +98,15 @@ module.exports = {
...
       [
         {
+<<<<<<< HEAD
           user_id: 3,
+=======
+<<<<<<< HEAD
+          user_id: 2,
+=======
+          user_id: 3,
+>>>>>>> 6da50f4 (* feat(profile) create profile feature (#37))
```
Comment (Niyobuzima) [inline src/database/seeders/20230326221727-demo-cogito.js]: and this

File: src/database/seeders/20230326221727-demo-cogito.js [hunk 7]
```diff
@@ -98,7 +98,15 @@ module.exports = {
       'wishlists',
       [
         {
+<<<<<<< HEAD
```
Comment (Niyobuzima) [inline src/database/seeders/20230326221727-demo-cogito.js]: delete this line

**Review by Niyobuzima — APPROVED** [review]

**Review by Charlesntwari — APPROVED** [review]

**Review by EvelyneUmubyeyi — APPROVED** [review]

**Review by dushimeemma — APPROVED** [review]
Good job 🥇

**Pull request merged**